dataset/get-metadata.go

```diff
@@ -92,6 +101,39 @@ func getEditMetadataHandler(w http.ResponseWriter, req *http.Request, dc Dataset
 
 }
 
+func getLatestPublishedVersionDimensions(ctx context.Context, w http.ResponseWriter, req *http.Request, dc DatasetClient, userAccessToken, collectionID, latestVersionURL string) []datasetclient.VersionDimension {
+	datasetID, editionID, versionID, err := getIDsFromURL(latestVersionURL)
+	if err != nil {
+		log.Event(ctx, "failed to parse latest version url", log.Error(err))
+		return []datasetclient.VersionDimension{}
+	}
+
+	latestPublishedVersion, err := dc.GetVersion(ctx, userAccessToken, "", "", collectionID, datasetID, editionID, versionID)
+	if err != nil {
+		log.Event(ctx, "failed Get latest published version details", log.Error(err))
+		setErrorStatusCode(req, w, err, datasetID)
+		return []datasetclient.VersionDimension{}
+	}
+
+	return latestPublishedVersion.Dimensions
+}
+
+func getIDsFromURL(URL string) (datasetID, editionID, versionID string, err error) {
```

Comment:
It would be worth writing a couple of unit tests for this function

Reply:
👍